### PR TITLE
Merge dotnet:master to dotnet:dev/apple_crypto (and add lost function)

### PR DESCRIPTION
**Pull request by bartonjs**
Merges recent merge conflicts from master to dev/apple_crypto, and the master PR noticed that there was a native method that got lost somewhere in the pipeline.

### CODE REVIEW

**Comment by bartonjs** [comment]
@stephentoub Mind giving https://github.com/dotnet/corefx/pull/17064/commits/c0e2ca30af981aa4fb07e6a5e180b93644602605 a look?

**Comment by bartonjs** [comment]
@dotnet-bot Test OuterLoop OSX10.12 Debug x64 please
@dotnet-bot Test OuterLoop OSX10.12 Release x64 please

File: src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_keychain.cpp [hunk 1]
```diff
@@ -208,6 +208,95 @@ static OSStatus DeleteInKeychain(CFTypeRef needle, SecKeychainRef haystack)
 }
 
 extern "C" int32_t
+AppleCryptoNative_X509StoreAddCertificate(CFTypeRef certOrIdentity, SecKeychainRef keychain, int32_t* pOSStatus)
+{
+    if (certOrIdentity == nullptr || keychain == nullptr)
```
Comment (stephentoub) [inline src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_keychain.cpp]: Nit: `|| pOSStatus == nullptr`?

File: src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_keychain.cpp [hunk 2]
```diff
@@ -208,6 +208,95 @@ static OSStatus DeleteInKeychain(CFTypeRef needle, SecKeychainRef haystack)
...
+        status = SecIdentityCopyCertificate(identity, &cert);
+
+        if (status == noErr)
+        {
+            status = SecIdentityCopyPrivateKey(identity, &privateKey);
+        }
+    }
+    else
+    {
+        return -1;
```
Comment (stephentoub) [inline src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_keychain.cpp]: It looks like pOSStatus is declared as out in the C# code... we should make sure we've initialized it on all return paths.

**Comment by bartonjs** [comment]
@dotnet-bot Test OuterLoop OSX10.12 Debug x64 please
@dotnet-bot Test OuterLoop OSX10.12 Release x64 please

(New commit to treat pOSStatus properly)

**Pull request merged**